src/wp-includes/media.php

```diff
@@ -1903,17 +1903,50 @@ function wp_filter_content_tags( $content, $context = null ) {
 	return $content;
 }
 
+/**
+ * Adds a fallback script to replace images in modern mime types with alternative mime types on older browsers.
+ *
+ * @since 6.1.0
+ *
+ * @global bool $_wp_image_mime_fallback_should_load
+ */
+function wp_print_image_mime_fallback_script() {
+	global $_wp_image_mime_fallback_should_load;
+
+	// Bail early if an image has no additional mime.
+	if ( true !== $_wp_image_mime_fallback_should_load ) {
+		return;
+	}
+
+	$suffix   = wp_scripts_get_suffix();
+	$version  = 'ver=' . get_bloginfo( 'version' );
+	$settings = array(
+		'restApi'                 => esc_url_raw( trailingslashit( get_rest_url() ) ),
+		/** This filter is documented in wp-includes/class.wp-scripts.php */
+		'imageMimeFallbackScript' => apply_filters( 'script_loader_src', includes_url( "js/wp-image-mime-fallback$suffix.js?$version" ), 'wp-image-mime-fallback' ),
+	);
+
+	wp_print_inline_script_tag(
+		sprintf( 'window._wpImageMimeFallbackSettings = %s;', wp_json_encode( $settings ) ) . "\n" .
+			file_get_contents( sprintf( ABSPATH . WPINC . '/js/wp-image-mime-fallback-loader' . $suffix . '.js' ) )
```

Comment:
The `sprintf()` call inside `file_get_contents()` seems unnecessary.

More generally, I'm a little uneasy about assuming the file exists and is readable. More specifically, that this is not handled. When this happens in JS, the console gets a 404. In this case, an`E_WARNING` will be generated.

However, on failure, what happens when `window._wpImageMimeFallbackSettings = wp_json_encode( $settings ) . "\n" . false`? What _should_ happen in this case? How should Core respond in this function, and should Core output anything to the user/log?

Reply:
@costdev FWIW, core script files like this can be assumed to exist and be readable. The code here entirely follows existing WordPress core code, namely the `_print_emoji_detection_script()` function.

If we couldn't assume that core files exist, I think we would have a lot of other more severe problems, so my vote here would be to go with what is already established in core.